oligo_designer_toolsuite/oligo_specificity_filter/_filter_blastn.py

```diff
@@ -187,15 +217,186 @@ def _find_hits(
+            "query_sequence",
+            "reference_start",
+            "reference_end",
+            "reference_sequence",
+            "reference_strand"
+        ]
+        if not all(field in self.names_search_output for field in required_fields):
+            raise ValueError(
+                f"Some of the required fields {required_fields} are missing in the search results."
+            )
+        
+        # set the positions to a 0-based index
+        references_plus = table_hits[table_hits["reference_strand"] == "plus"]
+        references_plus.loc[:, "query_start"] = references_plus["query_start"] - 1
+        references_plus.loc[:, "reference_start"] = references_plus["reference_start"] - 1
+        references_minus = table_hits[table_hits["reference_strand"] == "minus"]
+        references_minus.loc[:, "query_start"] = references_minus["query_start"] - 1
+        references_minus.loc[:, "reference_end"] = references_minus["reference_end"] - 1
+        table_hits = pd.concat([references_plus, references_minus])
+        table_hits.sort_index(inplace=True)  # restore the intial indexes
+
+        # create bed file for the plus strand
+        bed_plus = pd.DataFrame(
+            columns=["chr", "start", "end", "name", "score", "strand"],
+            index=references_plus.index,
+        )
+        bed_plus["chr"] = references_plus["reference"]
+        bed_plus["start"] = references_plus["reference_start"] - references_plus["query_start"]
+        bed_plus["end"] = references_plus["reference_end"] + (
+            references_plus["query_length"] - references_plus["query_end"]
+        )
+        bed_plus["name"] = references_plus["query"]
+        bed_plus["score"] = 0
+        bed_plus["strand"] = "+"
+
+        # create bed file for the minus strand
+        bed_minus = pd.DataFrame(
+            columns=["chr", "start", "end", "name", "score", "strand"],
+            index=references_minus.index,
+        )
+        bed_minus["chr"] = references_minus["reference"]
+        bed_minus["start"] = references_minus["reference_end"] - (
+            references_minus["query_length"] - references_minus["query_end"]
+        )
+        bed_minus["end"] = references_minus["reference_start"] + references_minus["query_start"]
+        bed_minus["name"] = references_minus["query"]
+        bed_minus["score"] = 0
+        bed_minus["strand"] = "-"
+
+        # concatenate the bed files
+        bed = pd.concat([bed_plus, bed_minus])
+        # adjust for possible overflows (e.g. new coordinates are not included in the gene boundaries)
+        # additionally we store how muchpadding we have to do to have two seqeunces of the same length
+        bed["overflow_start"] = bed["start"].apply(lambda x: -x if x < 0 else 0)
+        bed["start"] = bed["start"].apply(lambda x: x if x >= 0 else 0)
+        # what about having the leght of a region as a method of the reference database?
+        records = SeqIO.index(file_reference, "fasta")
```

Comment:
I think I would turn it into a utils/sequence_processor function

Reply:
The problem here is that it is split into 2 parts and its extremely specific to this method, hence the code would't be reused anywhere else.